Sources/SwiftBuildSupport/PIFBuilder.swift

```diff
@@ -362,6 +362,35 @@ public final class PIFBuilder {
                                     env[key.rawValue] = value
```

Comment:
You should switch `env` here over to the `Environment` abstraction from `Basics` rather than using a raw `[String: String]` -- this will handle the platform path list entry separator for you, and has convenience methods for adding a path to the path list, so you don't need to deal with all the split/join business.

`Environment` is also particularly important on Windows because the casing of PATH or Path can vary in different environments and using a raw dictionary can lead to reading/writing the environment incorrectly.

Reply:
I suggest also adding:

```swift
extension EnvironmentKey {
    package static var libraryPath: Self {
        #if os(Windows)
        path
        #elseif canImport(Darwin)
        "DYLD_LIBRARY_PATH"
        #else
        "LD_LIBRARY_PATH"
        #endif
    }
}
```

in EnvironmentKey.swift to make the platform-specific library path environment variable name easily reusable.